src/main/client/type.c

```diff
@@ -1152,7 +1385,7 @@ static int set_rack_aware_config(as_config *conf, PyObject *config_dict)
 
     py_config_value = PyDict_GetItemString(config_dict, "rack_id");
     if (py_config_value) {
-        if (PyLong_Check(py_config_value)) {
+        if (PyLong_CheckExact(py_config_value)) {
```

Comment:
Why the exact check? I'm not sure but I don't think a sub class would cause issues.